cxx-qt-gen/src/cxx_qt.cpp

```diff
@@ -55,8 +56,16 @@ extern "C"
     if (self->isNull())
       return false;
 
-    self->data()->requestUpdate();
-    return true;
+    const auto ret = QMetaObject::invokeMethod(
+      self->data(), "requestUpdate", Qt::DirectConnection);
```

Comment:
`requestUpdate` is now removed from the CxxQObject. Is this a problem for this call?

Reply:
This is calling the `requestUpdate` that is generated in `gen_cpp` (it's a little confusing), so this is the one in the generation https://github.com/ahayzen-kdab/cxx-qt/blob/locking-5-lambda-queue/cxx-qt-gen/src/gen_cpp.rs#L773

Or this one in the tests https://github.com/ahayzen-kdab/cxx-qt/blob/locking-5-lambda-queue/cxx-qt-gen/test_outputs/basic_update_requester.cpp#L31

The idea is that

- The user has the RequestUpdater object in Rust (which can be moved across threads)
- They trigger request_update on the Rust side
- This then triggers requestUpdate on the C++ object
- This then calls runOnGUIThread and queues the updateState call
- The updateState calls then locks and goes back to Rust from the GUI thread

Reply:
Ah, okay, didn't see that :)
I think we're ready to merge now :+1: